test/systemtest/utils.py

```diff
@@ -0,0 +1,428 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import difflib
+import filecmp
+import inspect
+import locale
+import logging
+import os
+import re
+import select
+import shlex
+import signal
+import subprocess
+import time
+from pathlib import Path
+
+import serial
+
+log = logging.getLogger(__name__)
+
+
+class Process:
+    """Utility class used to spawn an interact with processes."""
+    def __init__(self,
+                 cmd,
+                 logdir,
+                 cwd=None,
+                 startup_done_expect=None,
+                 startup_timeout=None,
+                 default_filter_func=None):
+        """Creates Process instance.
```

Comment:
Another convention for doing this is to put the documentation for the constructor into the docstring of the class. I quite like that, because you don't have to write the "Make a foo" sentence, which always seems a bit silly to me :-) No strong feelings though.